### ISSUE TITLE
Syntax Highlighting not working

### ISSUE THREAD

**Issue by gernotpokorny** [joaompinto/asciidoctor-vscode#112]
Why [syntax highlightning](https://asciidoctor.org/docs/asciidoc-writers-guide/#listing-and-source-code-blocks) is not working in the vscode-extension? I tested it with the examples from the [documentation](https://asciidoctor.org/docs/asciidoc-writers-guide/#listing-and-source-code-blocks).

**Comment by andrew-veresov** [comment]
Same here:
```
= Setup
:source-highlighter: highlightjs

[source,xml]
----
<test>some string</sdf>
----
```

What did you do to get highlighting working properly?

**Comment by sersanchus** [comment]
It's not working for me neither. Why is this issue closed?

**Comment by MatteoCampinoti94** [comment]
If the syntax highlighting isn't working in the rendered pdf/HTML then the problem is either a missing highlighter executable or some other problem unrelated to the extension.

If you are referring to the extension's live preview, then remember that it is meant as a quick way to check your document, not the way it WILL appear.

That said, if you set `asciidoc.use_asciidoctor_js` to `false` and use the system-installed asciidoctor then the preview will be based on HTML produced by the actual converter. Remember though that it is still NOT the same as the final result, because of various issues with the way it is fed to asciidoctor that we can't resolve easily.

**Comment by MatteoCampinoti94** [comment]
Then the problem is most likely not with the extension. Try to generate the HTML manually and then open it in your browser to check whether syntax highlighting works.

**Comment by Foadsf** [comment]
Same issue here. Tried with Scilab, it works fine on AsciiDocFx and Atom (though they are very slow). vscode is very fast on rendering live preview but for some reason the syntax highlighting doesn't work.

**Comment by Isam-Ilyas29** [comment]
> If the syntax highlighting isn't working in the rendered pdf/HTML then the problem is either a missing highlighter executable or some other problem unrelated to the extension.
> 
> If you are referring to the extension's live preview, then remember that it is meant as a quick way to check your document, not the way it WILL appear.
> 
> That said, if you set `asciidoc.use_asciidoctor_js` to `false` and use the system-installed asciidoctor then the preview will be based on HTML produced by the actual converter. Remember though that it is still NOT the same as the final result, because of various issues with the way it is fed to asciidoctor that we can't resolve easily.

Yep this worked for me. I installed asciidoctor through gem and then manually added the path through the extension settings. However I am using the **rouge** syntax highlighter (not highlightjs). Fortunately I was planning on using rouge any-who.

**Comment by peter-jerry-ye** [comment]
Just in case this might help some people.
There might be a small line on the bottom of the preview that says: 

> Some content has been disabled in this document

What you need to do is to Change Preview Security Setting (either by clicking the line or the menu on top right when the preview tab is selected) to `disable` as the highlightjs is a client side rendering + it's a script, which means that vscode has to fetch the hightlightjs from cdn + execute the script which vscode finds not secure.
